tests/unit_tests/config/test_queue_config.py

```diff
@@ -83,22 +83,130 @@ def test_that_invalid_queue_option_raises_validation_error(
 @st.composite
 def memory_with_unit(draw):
     memory_value = draw(st.integers(min_value=1, max_value=10000))
-    unit = draw(st.sampled_from(["gb", "mb"]))
+    unit = draw(
+        st.sampled_from(["gb", "mb", "tb", "pb", "Kb", "Gb", "Mb", "Pb", "b", "B", ""])
+    )
     return f"{memory_value}{unit}"
 
 
 @pytest.mark.usefixtures("use_tmpdir", "set_site_config")
-@pytest.mark.parametrize("memory_with_unit_str", ["gb", "mb", "1 gb"])
-def test_that_invalid_memory_pr_job_raises_validation_error(memory_with_unit_str):
+@given(memory_with_unit())
+def test_supported_memory_units_to_realization_memory(
+    memory_with_unit,
+):
     filename = "config.ert"
     with open(filename, "w", encoding="utf-8") as f:
         f.write("NUM_REALIZATIONS 1\n")
-        f.write("QUEUE_SYSTEM TORQUE\n")
-        f.write(f"QUEUE_OPTION TORQUE MEMORY_PER_JOB {memory_with_unit_str}")
+        f.write(f"REALIZATION_MEMORY {memory_with_unit}\n")
+
+    assert ErtConfig.from_file(filename).queue_config.realization_memory > 0
+
+
+@pytest.mark.parametrize(
+    "memory_spec, expected_bytes",
+    [
+        ("1", 1),
+        ("1b", 1),
+        ("10b", 10),
+        ("10kb", 10 * 1024),
+        ("10mb", 10 * 1024**2),
+        ("10gb", 10 * 1024**3),
+        ("10Mb", 10 * 1024**2),
+        ("10Gb", 10 * 1024**3),
+        ("10Tb", 10 * 1024**4),
+        ("10Pb", 10 * 1024**5),
+    ],
+)
+def test_realization_memory_unit_support(memory_spec: str, expected_bytes):
```

Comment:
Should we use the tmp_dir fixture here, or is it in autouse?

Reply:
We should. Fixed.